### PR TITLE
7 comment post and relevant errros

### PR DESCRIPTION
**Pull request by crepsleyfox**
finally. is it ok to have the error handle for non existent username in the comments model or best to move it to the users model ?

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -54,6 +54,40 @@ describe("NC News Server", () => {
...
+        const newComment = { username: 'lurker', body: 'body of comment 123' }
+        return request(app)
+            .post('/api/articles/1000/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({ body}) => {
+            expect(body.message).toBe("Article Not Found");
+          });
+      });
+  })
```
Comment (popatre) [inline __tests__/app.test.js:90]: TEST: Status 400: Bad request, invalid article_id
Responds with error object including status and message.
E.g. POST /api/articles/notanumber/comments

File: __tests__/app.test.js [hunk 2]
```diff
@@ -170,4 +204,18 @@ describe("NC News Server", () => {
...
+    test("POST 201 : adds a comment to a specific article", () => {
+        const newComment = { username: 'lurker', body: 'body of comment 123' }
+        return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                const { comment } = body
+                expect(comment.author).toBe('lurker')
+                expect(comment.body).toBe('body of comment 123')
```
Comment (popatre) [inline __tests__/app.test.js:217]: Ideally you'd want to check for the other values on this object as well

**Review by popatre — CHANGES_REQUESTED** [review]
Just need a few things please:

- [ ] TEST: Status 400: Bad request, invalid article_id - E.g. POST /api/articles/notanumber/comments
- [ ] Check for the other values on the returned object as well to make sure you're getting back all the data you need

File: __tests__/app.test.js [hunk 3]
```diff
@@ -204,7 +212,7 @@ describe("NC News Server", () => {
         });
     });
   });
-  describe("POST /api/articles/:article_id/comments", () => {
+  describe.only("POST /api/articles/:article_id/comments", () => {
```
Comment (nc-verity-gregory) [inline __tests__/app.test.js:215]: Please don't leave these in, remove it and resubmit.

**Review by nc-verity-gregory — CHANGES_REQUESTED** [review]
There is a .only in your test file, I'm afraid I can't review this fully until it is gone and we know all tests are running.

File: endpoints.json [hunk 4]
```diff
@@ -77,5 +77,23 @@
         ]
       }
     }
+  },
+  "POST /api/articles/:article_id/comments": {
+    "description": "post a comment on a given article",
```
Comment (nc-jonathan-mcguire) [inline endpoints.json:82]: You'll thank me when we get to front end - but i would document here what the post body needs to look like i.e. what keys they need to send

**Review by nc-jonathan-mcguire — APPROVED** [review]
See my comment which you'll be grateful of when we get to frontend

**Pull request merged**